demos/coreHTTP/http_demo_s3_download.c

```diff
@@ -0,0 +1,729 @@
+                               const char * pIdentifier,
+                               void * pNetworkServerInfo,
+                               void * pNetworkCredentialInfo,
+                               const IotNetworkInterface_t * pNetworkInterface )
+{
+    /* The transport layer interface used by the HTTP Client library. */
+    TransportInterface_t xTransportInterface;
+    /* The network context for the transport layer interface. */
+    NetworkContext_t xNetworkContext = { 0 };
+    TransportSocketStatus_t xNetworkStatus;
+    BaseType_t xIsConnectionEstablished = pdFALSE;
+    /* HTTPS Client library return status. */
+    HTTPStatus_t xHTTPStatus = HTTPSuccess;
+
+    /* Upon return, pdPASS will indicate a successful demo execution.
+    * pdFAIL will indicate some failures occurred during execution. The
+    * user of this demo must check the logs for any failure codes. */
+    BaseType_t xDemoStatus = pdPASS;
+
+    /* Remove compiler warnings about unused parameters. */
+    ( void ) awsIotMqttMode;
+    ( void ) pIdentifier;
+    ( void ) pNetworkServerInfo;
+    ( void ) pNetworkCredentialInfo;
+    ( void ) pNetworkInterface;
+
+    /* The length of the path within the pre-signed URL. This variable is
+     * defined in order to store the length returned from parsing the URL, but
+     * it is unused. The path used for the requests in this demo needs all the
+     * query information following the location of the object, to the end of the
+     * S3 presigned URL. */
+    size_t pathLen = 0;
+
+    LogInfo( ( "HTTP Client Synchronous S3 download demo using pre-signed URL:\n%s",
+               democonfigS3_PRESIGNED_GET_URL ) );
+
+    /**************************** Connect. ******************************/
+
+    /* Establish TLS connection on top of TCP connection using Secure Sockets. */
+    if( xDemoStatus == pdPASS )
+    {
+        /* Attempt to connect to S3. If connection fails, retry after a timeout.
+         * The timeout value will be exponentially increased until either the
+         * maximum number of attempts or the maximum timeout value is reached.
+         * The function returns EXIT_FAILURE if the TCP connection cannot be
+         * established to the broker after the configured number of attempts. */
+        xDemoStatus = connectToServerWithBackoffRetries( prvConnectToServer,
+                                                         &xNetworkContext );
+
+        if( xDemoStatus == pdFAIL )
+        {
+            /* Log an error to indicate connection failure after all
+             * reconnect attempts are over. */
+            LogError( ( "Failed to connect to HTTP server %s.",
+                        pxServerHost ) );
+        }
+    }
+
+    /* Define the transport interface. */
+    if( xDemoStatus == pdPASS )
+    {
+        /* Set a flag indicating a TLS connection exists. This is done to
+         * disconnect if the loop exits before disconnection happens. */
```

Comment:
We can take out this loop comment from the demos.

Reply:
Thanks for catching that.